librz/asm/arch/evm/evmasm.c

```diff
@@ -0,0 +1,359 @@
+	} else if (!strcmp("sload", buf)) {
+		opbuf[0] = 0x54;
+	} else if (!strcmp("sstore", buf)) {
+		opbuf[0] = 0x55;
+	} else if (!strcmp("jump", buf)) {
+		opbuf[0] = 0x56;
+	} else if (!strcmp("jump", buf)) {
+		opbuf[0] = 0x56;
+	} else if (!strcmp("jumpi", buf)) {
+		opbuf[0] = 0x57;
+	} else if (!strcmp("pc", buf)) {
+		opbuf[0] = 0x58;
+	} else if (!strcmp("msize", buf)) {
+		opbuf[0] = 0x59;
+	} else if (!strcmp("gas", buf)) {
+		opbuf[0] = 0x5a;
+	} else if (!strcmp("jumpdest", buf)) {
+		opbuf[0] = 0x5b;
+	} else if (!strncmp("push", buf, 3)) {
+		char out[100];
+		int number = atoi(buf+4);
+		switch (number) {
+			case 1:
+			len = 2;
+			opbuf[0] = 0x60;
+			break;
+		case 2:
+			len = 3;
+			opbuf[0] = 0x61;
+			break;
+		case 3:
+			len = 4;
+			opbuf[0] = 0x62;
+			break;
+		case 4:
+			len = 5;
+			opbuf[0] = 0x63;
+			break;
+		case 5:
+			len = 6;
+			opbuf[0] = 0x64;
+			break;
+		case 6:
+			len = 7;
+			opbuf[0] = 0x65;
+			break;
+		case 7:
+			len = 8;
+			opbuf[0] = 0x66;
+			break;
+		case 8:
+			len = 9;
+			opbuf[0] = 0x67;
+			break;
+		case 9:
+			len = 10;
+			opbuf[0] = 0x68;
+			break;
+		case 10:
+			len = 11;
+			opbuf[0] = 0x69;
+			break;
+		case 11:
+			len = 12;
+			opbuf[0] = 0x6a;
+			break;
+		case 12:
+			len = 13;
+			opbuf[0] = 0x6b;
+			break;
+		case 13:
+			len = 14;
+			opbuf[0] = 0x6c;
+			break;
+		case 14:
+			len = 15;
+			opbuf[0] = 0x6d;
+			break;
+		case 15:
+			len = 16;
+			opbuf[0] = 0x6e;
+			break;
+		case 16:
+			len = 17;
+			opbuf[0] = 0x6f;
+			break;
+		case 17:
+			len = 18;
+			opbuf[0] = 0x70;
+			break;
+				case 18:
+			len = 19;
+			opbuf[0] = 0x71;
+			break;
+		case 19:
+			len = 20;
+			opbuf[0] = 0x72;
+			break;
+		case 20:
+			len = 21;
+			opbuf[0] = 0x73;
+			break;
+		case 21:
+			len = 22;
+			opbuf[0] = 0x74;
+			break;
+		case 22:
+			len = 23;
+			opbuf[0] = 0x75;
+			break;
+		case 23:
+			len = 24;
+			opbuf[0] = 0x76;
+			break;
+		case 24:
+			len = 25;
+			opbuf[0] = 0x77;
+			break;
+		case 25:
+			len = 26;
+			opbuf[0] = 0x78;
+			break;
+		case 26:
+			len = 27;
+			opbuf[0] = 0x79;
+			break;
+		case 27:
+			len = 28;
+			opbuf[0] = 0x7a;
+			break;
+		case 28:
+			len = 29;
+			opbuf[0] = 0x7b;
+			break;
+		case 29:
+			len = 30;
+			opbuf[0] = 0x7c;
+			break;
+		case 30:
+			len = 31;
+			opbuf[0] = 0x7d;
+			break;
+		case 31:
+			len = 32;
+			opbuf[0] = 0x7e;
+			break;
+		case 32:
+			len = 33;
+			opbuf[0] = 0x7f;
+			break;
+		}
+		int opbuf_len;
+	    	char two_chars_str[3];
+	    	if (number <= 9) {
+			for (int j = 8, i = 0; i < len-1; j += 2, i ++) {
+				two_chars_str[0] = buf[j];
+				two_chars_str[1] = buf[j+1];
+				two_chars_str[2] = '\0';
+				opbuf[i+1] = strtol(two_chars_str, NULL, 16);
+			}
+	    	} else if (number > 9) {
+			for (int j = 9, i = 0; i < len-1; j += 2, i ++) {
+				two_chars_str[0] = buf[j];
+				two_chars_str[1] = buf[j+1];
+				two_chars_str[2] = '\0';
+				opbuf[i+1] = strtol(two_chars_str, NULL, 16);
+			}
```

Comment:
there are methods in rizin to convert hex to bytes. also you don't really need to use strtol for doing that.